extensions/ql-vscode/src/remote-queries/remote-queries-monitor.ts

```diff
@@ -0,0 +1,97 @@
+import * as vscode from 'vscode';
+import { Credentials } from '../authentication';
+import { Logger } from '../logging';
+import { showAndLogErrorMessage } from '../helpers';
+import { RemoteQuery } from './remote-query';
+import { RemoteQueryWorkflowResult } from './remote-query-workflow-result';
+
+export class RemoteQueriesMonitor {
+  private static readonly twoDaysInSeconds = 86400;
+
+  constructor(
+    private readonly extensionContext: vscode.ExtensionContext,
+    private readonly logger: Logger
+  ) {
+  }
+
+  public async monitorQuery(
+    remoteQuery: RemoteQuery,
+    cancellationToken: vscode.CancellationToken
+  ): Promise<RemoteQueryWorkflowResult> {
+    const credentials = await Credentials.initialize(this.extensionContext);
+
+    if (!credentials) {
+      await showAndLogErrorMessage('Error authenticating with GitHub');
+    }
+
+    let attemptCount = 0;
+
+    const octokit = await credentials.getOctokit();
+
+    while (attemptCount <= RemoteQueriesMonitor.twoDaysInSeconds) {
+      if (cancellationToken && cancellationToken.isCancellationRequested) {
+        return { status: 'Cancelled' };
+      }
+
+      const workflowRun = await octokit.rest.actions.getWorkflowRun({
+        owner: remoteQuery.controllerRepository.owner,
+        repo: remoteQuery.controllerRepository.name,
+        run_id: remoteQuery.actionsWorkflowRunId
+      });
+
+      if (this.areStringsSame(workflowRun.data.status, 'completed')) {
+        if (this.areStringsSame(workflowRun.data.conclusion, 'success')) {
+          return { status: 'CompletedSuccessfully' };
+        } else {
+          const error = this.getWorkflowError(workflowRun.data.conclusion);
+          return { status: 'CompletedUnsuccessfully', error };
+        }
+      }
+
+      await this.sleep(1000);
```

Comment:
This feels like it is checking too often. There might be an API rate limit that gets hit here. We need to be careful. Maybe bump it to every 3 or 5 seconds?

Reply:
Thank you - this is great feedback! I've updated to 5secs.